src/validators/object.ts

```diff
@@ -173,36 +176,60 @@ export abstract class ObjectValidator<T extends ObjectSchema = never, O = never>
           }
         }
 
+        // Serde
+        let serdeStr: string
+        let deriveMacro: string[] | undefined = undefined
+
+        if (this.deriveMacro !== undefined) {
+          const deriveMacroStr = this.deriveMacro.join(', ')
+          serdeStr = `#[derive(${deriveMacroStr})]\n`
+          deriveMacro = this.deriveMacro
+        } else if (options.deriveMacro !== undefined) {
```

Comment:
Are you not setting ```this.deriveMacro = options?.deriveMacro``` in the constructor and you are setting this.deriveMacro if options.deriveMacro !== undefined, don't think this code block makes sense.

Reply:
There's the deriveMacro from "options" which would be the the call toString() where the parrent can pass in stuff.
And then there's the constructor/class field, which takes priority. (So manually writing it).